artcommon/artcommonlib/assembly.py

```
@@ -232,6 +232,31 @@ def _merger(a, b):
+    Check whether an 'until' expiry date has passed.
+
+    Arg(s):
+        until_str (str): ISO 8601 date string (YYYY-MM-DD) from an 'until' field.
+        context (str): Human-readable description of the override (for error messages).
+    Return Value(s):
+        str | None: Error message if expired, None if still valid.
+    """
+    try:
+        until_date = date.fromisoformat(until_str)
+    except ValueError:
+        raise ValueError(f"Invalid 'until' date format in {context}: {until_str!r}. Expected YYYY-MM-DD.")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)